src/librbd/ImageWatcher.cc

```diff
@@ -850,6 +866,33 @@ Context *ImageWatcher<I>::prepare_unquiesce_request(const AsyncRequestId &reques
     });
 }
 
+template <typename I>
+void ImageWatcher<I>::cancel_quiesce_requests(Context *on_finish) {
+  {
+    std::unique_lock l{m_async_request_lock};
+    for (auto it = m_async_requests.begin(); it != m_async_requests.end(); ) {
+      if (it->second.second == nullptr) {
+        // Quiesce notify request
+        mark_async_request_complete(it->first, 0);
```

Comment:
`cancel_async_requests()` was alread invoked by this time -- should this just move to that method?

Reply:
`cancel_async_requests` may be called not only on unregister_watcher, and then it seems we don't want the quiesce requests cancelled.

Also `cancel_async_requests` is called before `Watcher::unregister_watch` so I suppose there is a small window that a new quiesce request comes.

I think I could add a param (`bool cancel_quiesce`) to `cancel_async_requests` and then move `cancel_async_requests` call after `unregister_watch`, if you like this more than the current approach...

Reply:
Eh, probably not worth it then. Still interested in your thoughts about having `TaskFinisher` just complete the tasks w/ `-ECANCELED`, though.